### PR TITLE
AMD

### PR DESCRIPTION
**Pull request by greggman**
See #143 

This PR adds basic WebGL2 support (it runs, adds WebGL2 VAO support, otherwise lots of features still TBD), and switches the module system to use AMD and the builder to use Webpack+Babel so you can use ES6 features and compile down to ES5/IE

The chrome extension works in both release and debug mode and embed.js seems to work in both release and debug mode. Also tested firefox in release. Safari I can't test as the extension no longer works even without these changes, see #145 

A few caveats: Because the debug mode uses require.js there are issues with using debug mode on pages that use require.js themselves. There is a workaround. Non-debug mode does not have that issue. See readme.md

Another issue is in debug mode things are finicky in that it's possible `window.onload` could fire
before `getContext` has been wrapped. At the moment it seems to be working but just in case, for those cases you can (debug mode with embed.js only) you can wait for `gliready` instead of load. In other words assuming you have some code like this

    window.onload = runMyApp;

You'd change it to

    window.addEventListener('gliready', runMyApp);

Again, only when using embed.js in debug mode. In non debug mode there's no issue. Also mentioned in readme

### CODE REVIEW

**Comment by benvanik** [comment]
Still reading through this, but it looks fantastic Gregg!

File: core/ui/tabs/trace/TraceListing.js [hunk 1]
```diff
@@ -100,7 +111,7 @@
         if (call.error) {
             el.className += " trace-call-error";
 
-            var errorString = gli.info.enumToString(call.error);
+            var errorString = infos.enumToString(call.error);
```
Comment (benvanik) [inline core/ui/tabs/trace/TraceListing.js]: info*?

**Comment by benvanik** [comment]
Would you like to merge this soon, or are you planning to continue updating this PR? There were a few TODO's in there (like the vertex array not refreshing) but if you're planning on adding more to this PR that's fine.

File: core/ui/tabs/trace/TraceListing.js (hunk 1 above)
Comment (greggman) [inline core/ui/tabs/trace/TraceListing.js]: fixed

**Comment by greggman** [comment]
I'd like to merge this if that's ok. I don't think there's any loss in functionality and the AMD changes are pretty big and would make it harder to merge other changes later if someone happened to work on the non AMD version.

I have another PR waiting and more coming.

As for the TODOs I scanned over them and I don't see any I wrote. They're all copied from code I cribbed. I think the refresh comment you're referring to is copied from `BuffersTab.js`

**Pull request merged**